### PR TITLE
Expose GetCompactionReasonString() publicly

### PR DESCRIPTION
**Pull request by thepulkitagarwal**
For Issue #6471 

Add function `GetStringFromCompactionReason(...)` to get a string value from compaction reason, similar to compression type in `listener.h`.

### CODE REVIEW

File: db/compaction/compaction.h [hunk 1]
```diff
@@ -281,6 +281,8 @@ class Compaction {
 
   int GetInputBaseLevel() const;
 
+  const char* GetCompactionReasonString() const;
```
Comment (ajkr) [inline db/compaction/compaction.h]: In order to be publicly visible (i.e., usable by RocksDB users), the function needs to be declared in a header file under "include/" directory. For example take a look at how `GetStringFromCompressionType()` is declared: https://github.com/facebook/rocksdb/blob/e171a219d592b1bda9b654db126f04d4d834a815/include/rocksdb/convenience.h#L296-L297.

File: options/options_helper.cc [hunk 2]
```diff
@@ -253,6 +254,25 @@ std::unordered_map<std::string, CompressionType>
         {"kZSTD", kZSTD},
         {"kZSTDNotFinalCompression", kZSTDNotFinalCompression},
         {"kDisableCompressionOption", kDisableCompressionOption}};
+
+std::unordered_map<std::string, CompactionReason>
```
Comment (ajkr) [inline options/options_helper.cc]: I think we shouldn't follow `GetStringFromCompressionType()`'s implementation pattern - just following the declaration pattern (declaring `GetStringFromCompactionReason()` in a file under an "include/" directory) is enough. `CompactionReason` is not an option so its logic doesn't really belong here. We also don't do lookup based on string so the map isn't necessary.

Also if you want to make analogous changes for the other enums in "include/rocksdb/listener.h", that'd be helpful (e.g., `TableFileCreationReason`).

File: db/compaction/compaction.h (hunk 1 above)
Comment (ajkr) [reply db/compaction/compaction.h]: Sorry I probably shouldn't have provided this example here, as the declaration itself is not substantial. Meanwhile, the example function looks too similar to what our function does so it's tempting to follow its implementation pattern.

File: options/options_helper.cc (hunk 2 above)
Comment (thepulkitagarwal) [reply options/options_helper.cc]: I'm not sure where it should be defined. I thought of putting the definition in `compaction.h`, but then the other enums' definitions wouldn't fit in there. What file would you suggest for that?

Comment (ajkr) [reply options/options_helper.cc]: Looks like everything declared in `include/rocksdb/listener.h` so far is defined inline. We could either continue that pattern or create a `db/listener.cc` file for these definitions. Or, if you have the declarations in `include/rocksdb/convenience.h`, then the definitions should go in `db/convenience.cc`.

**Comment by cbi42** [comment]
Implemented in #11778

**Pull request closed without merging**